### PR TITLE
REGISTRY-3247 - Fix for clear all functionality

### PR DESCRIPTION
**Pull request by mushthaq33**

### CODE REVIEW

File: modules/es-extensions/store/extensions/app/greg-apis/apis/notification.jag [hunk 1]
```diff
@@ -61,17 +63,22 @@ require('/modules/store.js').exec(function(ctx) {
...
-            } catch (e){
-              var message= {'error':'Not supported'};
-              print(message);
+          if (matcher.match(SUB_ITEM)) {
+              var idObj = new org.apache.axis2.databinding.types.URI(options.id);
+              try{
+                  taskOperationService.start(idObj);
+                  taskOperationService.complete(idObj, "<WorkResponse>true</WorkResponse>");
+              } catch (e){
+                  message= {'error':'Not supported'};
```
Comment (daneshk) [inline modules/es-extensions/store/extensions/app/greg-apis/apis/notification.jag]: formatting issue. please check

Comment (mushthaq33) [inline modules/es-extensions/store/extensions/app/greg-apis/apis/notification.jag]: Fixed with https://github.com/wso2/product-greg/pull/851

File: modules/es-extensions/publisher/extensions/app/greg-publisher-defaults/themes/default/js/greg-subscriptions-api.js [hunk 2]
```diff
@@ -160,8 +160,12 @@ $(function() {
             contentType: 'application/json',
             success: function(data) {
                 if (data.error == null) {
+                    var updatedNotiCount = notiCount - 1;
                     element.parent().remove();
-                    $('#notificationCount').html((notiCount - 1));
+                    $('#notificationCount').html(updatedNotiCount);
+                    if(updatedNotiCount == 0){
```
Comment (splinter) [inline modules/es-extensions/publisher/extensions/app/greg-publisher-defaults/themes/default/js/greg-subscriptions-api.js]: Code Formatting issue

File: modules/es-extensions/store/extensions/app/greg-store-defaults/themes/store/js/greg-subscriptions-api.js [hunk 3]
```diff
@@ -91,15 +91,66 @@ $(function() {
             contentType: 'application/json',
             success: function(data) {
                 if (data.error == null) {
+                    var updatedNotiCount = notiCount - 1;
                     element.parent().remove();
-                    $('#notificationCount').html((notiCount - 1));
+                    $('#notificationCount').html(updatedNotiCount);
+                    if(updatedNotiCount == 0){
```
Comment (splinter) [inline modules/es-extensions/store/extensions/app/greg-store-defaults/themes/store/js/greg-subscriptions-api.js]: Code Formatting issue

File: modules/es-extensions/publisher/extensions/app/greg-apis/apis/notification.jag [hunk 4]
```diff
@@ -61,17 +63,22 @@ require('/modules/publisher.js').exec(function(ctx) {
...
-              taskOperationService.complete(new org.apache.axis2.databinding.types.URI(options.id), "<WorkResponse>true</WorkResponse>");
-            } catch (e){
-              var message= {'error':'Not supported'};
-              print(message);
+          if (matcher.match(SUB_ITEM)) {
+              var idObj = new org.apache.axis2.databinding.types.URI(options.id);
+              try{
+                  taskOperationService.start(idObj);
+                  taskOperationService.complete(idObj, "<WorkResponse>true</WorkResponse>");
+              } catch (e){
```
Comment (splinter) [inline modules/es-extensions/publisher/extensions/app/greg-apis/apis/notification.jag]: Code Formatting issue

File: modules/es-extensions/publisher/extensions/app/greg-publisher-defaults/themes/default/js/greg-subscriptions-api.js (hunk 2 above)
Comment (splinter) [inline modules/es-extensions/publisher/extensions/app/greg-publisher-defaults/themes/default/js/greg-subscriptions-api.js]: Code Formatting issue

File: modules/es-extensions/publisher/modules/greg-publisher-api.js [hunk 5]
```diff
@@ -267,6 +270,35 @@ var gregAPI = {};
...
+        queryInput.setSimpleQueryCategory(
+            org.wso2.carbon.humantask.client.api.types.TSimpleQueryCategory.ASSIGNED_TO_ME);
+        var resultSet = taskOperationService.simpleQuery(queryInput);
+        var rows = resultSet.getRow();
+        if (rows) {
+            for (var i = 0; i < rows.length; i++) {
+                var row =  rows[i];
+                var id = String(row.getId());
+                var idObj = new org.apache.axis2.databinding.types.URI(id);
+                try{
```
Comment (splinter) [inline modules/es-extensions/publisher/modules/greg-publisher-api.js]: Code Formatting issue

File: modules/es-extensions/publisher/modules/greg-publisher-api.js [hunk 6]
```diff
@@ -267,6 +270,35 @@ var gregAPI = {};
...
+                    taskOperationService.start(idObj);
+                    taskOperationService.complete(idObj, "<WorkResponse>true</WorkResponse>");
+                } catch (e){
```
Comment (splinter) [inline modules/es-extensions/publisher/modules/greg-publisher-api.js]: Code Formatting issue

File: modules/es-extensions/store/extensions/app/greg-apis/apis/notification.jag [hunk 7]
```diff
@@ -61,17 +63,22 @@ require('/modules/store.js').exec(function(ctx) {
...
-              taskOperationService.complete(new org.apache.axis2.databinding.types.URI(options.id), "<WorkResponse>true</WorkResponse>");
-            } catch (e){
-              var message= {'error':'Not supported'};
-              print(message);
+          if (matcher.match(SUB_ITEM)) {
+              var idObj = new org.apache.axis2.databinding.types.URI(options.id);
+              try{
+                  taskOperationService.start(idObj);
+                  taskOperationService.complete(idObj, "<WorkResponse>true</WorkResponse>");
+              } catch (e){
```
Comment (splinter) [inline modules/es-extensions/store/extensions/app/greg-apis/apis/notification.jag]: Code Formatting issue

File: modules/es-extensions/store/modules/greg-publisher-api.js [hunk 8]
```diff
@@ -248,6 +251,35 @@ var gregAPI = {};
...
+        var rows = resultSet.getRow();
+        if (rows) {
+            for (var i = 0; i < rows.length; i++) {
+                var row =  rows[i];
+                var id = String(row.getId());
+                var idObj = new org.apache.axis2.databinding.types.URI(id);
+                try{
+                    taskOperationService.start(idObj);
+                    taskOperationService.complete(idObj, "<WorkResponse>true</WorkResponse>");
+                } catch (e){
```
Comment (splinter) [inline modules/es-extensions/store/modules/greg-publisher-api.js]: Code Formatting issue

File: modules/es-extensions/store/modules/greg-publisher-api.js [hunk 9]
```diff
@@ -248,6 +251,35 @@ var gregAPI = {};
...
+        queryInput.setSimpleQueryCategory(
+            org.wso2.carbon.humantask.client.api.types.TSimpleQueryCategory.ASSIGNED_TO_ME);
+        var resultSet = taskOperationService.simpleQuery(queryInput);
+        var rows = resultSet.getRow();
+        if (rows) {
+            for (var i = 0; i < rows.length; i++) {
+                var row =  rows[i];
+                var id = String(row.getId());
+                var idObj = new org.apache.axis2.databinding.types.URI(id);
+                try{
```
Comment (splinter) [inline modules/es-extensions/store/modules/greg-publisher-api.js]: Code Formatting issue

File: modules/es-extensions/store/extensions/app/greg-apis/apis/notification.jag [hunk 10]
```diff
@@ -61,17 +63,22 @@ require('/modules/store.js').exec(function(ctx) {
...
       case 'DELETE':
-          try{
-              taskOperationService.start(new org.apache.axis2.databinding.types.URI(options.id));
-              taskOperationService.complete(new org.apache.axis2.databinding.types.URI(options.id), "<WorkResponse>true</WorkResponse>");
-            } catch (e){
-              var message= {'error':'Not supported'};
-              print(message);
+          if (matcher.match(SUB_ITEM)) {
+              var idObj = new org.apache.axis2.databinding.types.URI(options.id);
+              try{
```
Comment (splinter) [inline modules/es-extensions/store/extensions/app/greg-apis/apis/notification.jag]: Code Formatting issue

File: modules/es-extensions/publisher/extensions/app/greg-apis/apis/notification.jag [hunk 11]
```diff
@@ -61,17 +63,22 @@ require('/modules/publisher.js').exec(function(ctx) {
...
       case 'DELETE':
-          try{
-              taskOperationService.start(new org.apache.axis2.databinding.types.URI(options.id));
-              taskOperationService.complete(new org.apache.axis2.databinding.types.URI(options.id), "<WorkResponse>true</WorkResponse>");
-            } catch (e){
-              var message= {'error':'Not supported'};
-              print(message);
+          if (matcher.match(SUB_ITEM)) {
+              var idObj = new org.apache.axis2.databinding.types.URI(options.id);
+              try{
```
Comment (splinter) [inline modules/es-extensions/publisher/extensions/app/greg-apis/apis/notification.jag]: Code Formatting issue

**Comment by mushthaq33** [comment]
All the code formatting issue are resolved in the following commit - 63fa288

File: modules/es-extensions/publisher/extensions/app/greg-publisher-defaults/themes/default/js/greg-subscriptions-api.js (hunk 2 above)
Comment (mushthaq33) [inline modules/es-extensions/publisher/extensions/app/greg-publisher-defaults/themes/default/js/greg-subscriptions-api.js]: Thanks for the comment. Resolved in the following commit - 63fa288

File: modules/es-extensions/store/extensions/app/greg-store-defaults/themes/store/js/greg-subscriptions-api.js (hunk 3 above)
Comment (mushthaq33) [inline modules/es-extensions/store/extensions/app/greg-store-defaults/themes/store/js/greg-subscriptions-api.js]: Thanks for the comment. Resolved in the following commit - 63fa288

File: modules/es-extensions/publisher/extensions/app/greg-apis/apis/notification.jag (hunk 4 above)
Comment (mushthaq33) [inline modules/es-extensions/publisher/extensions/app/greg-apis/apis/notification.jag]: Thanks for the comment. Resolved in the following commit - 63fa288

File: modules/es-extensions/publisher/modules/greg-publisher-api.js (hunk 5 above)
Comment (mushthaq33) [inline modules/es-extensions/publisher/modules/greg-publisher-api.js]: Thanks for the comment. Resolved in the following commit - 63fa288

File: modules/es-extensions/publisher/modules/greg-publisher-api.js (hunk 6 above)
Comment (mushthaq33) [inline modules/es-extensions/publisher/modules/greg-publisher-api.js]: Thanks for the comment. Resolved in the following commit - 63fa288

File: modules/es-extensions/store/extensions/app/greg-apis/apis/notification.jag (hunk 7 above)
Comment (mushthaq33) [inline modules/es-extensions/store/extensions/app/greg-apis/apis/notification.jag]: Thanks for the comment. Resolved in the following commit - 63fa288

File: modules/es-extensions/store/modules/greg-publisher-api.js (hunk 8 above)
Comment (mushthaq33) [inline modules/es-extensions/store/modules/greg-publisher-api.js]: Thanks for the comment. Resolved in the following commit - 63fa288

File: modules/es-extensions/store/modules/greg-publisher-api.js (hunk 9 above)
Comment (mushthaq33) [inline modules/es-extensions/store/modules/greg-publisher-api.js]: Thanks for the comment. Resolved in the following commit - 63fa288

File: modules/es-extensions/store/extensions/app/greg-apis/apis/notification.jag (hunk 10 above)
Comment (mushthaq33) [inline modules/es-extensions/store/extensions/app/greg-apis/apis/notification.jag]: Thanks for the comment. Resolved in the following commit - 63fa288

File: modules/es-extensions/publisher/extensions/app/greg-apis/apis/notification.jag (hunk 11 above)
Comment (mushthaq33) [inline modules/es-extensions/publisher/extensions/app/greg-apis/apis/notification.jag]: Thanks for the comment. Resolved in the following commit - 63fa288

**Pull request merged**